Tags instead ''.

Remski01 opened the pull request:
Hello. 

I useing MyWarp for an long time and after editing the localization i got some realy smal things that got annoying because its MUST have ' ' and maybay you can change that with tags like %warp %player etc, instead ''{0}'' 

Also i dont get it why some messages MUST are &c only

So if it can please add that. if not then not :p Its just an idea :)

TheE committed:
Fixed modify permission.

TheE committed:
Fixed BlockSafety.

TheE committed:
Two characters can make a difference.

TheE committed:
Made MessageFormat use the local of the loaded resource-bundle.

TheE committed:
Get the limit directly instead of using custom methods for each number.

TheE committed:
Added German translation and fixed a lot of typos/spelling errors.
Special thanks to 'Monsterkueken' for her help.

TheE committed:
Fixed value-order for (un)invitation-messages.

TheE committed:
Fixed missing warp.signs.created-successful message.

TheE committed:
Quotation marks are not needed here.

TheE committed:
BRAKING CHANGE: Made limits work per world or per multiple worlds
While this change could have been done without braking existing setups,
I think it is the perfect point to fix some ugly stuff:
configured-limits now have their own configuration-section, permissions
are in line with the timer's 'disobey' syntax.

The new concept should still be easy to use for global setups, but also
covers more complex approaches. The default-limit is forced to be
global, other limits can be configured to match only warps on certain
worlds. Players can disobey limits per type and world, per world only,
per type only, and completely. All permissions use internal parent/child
structure so there is no overhead when checking on creation.

TheE committed:
Pushed bukkit to 1.7.2-R0.1-SNAPSHOT in pom.xml

TheE committed:
Added 1.7 blocks to warp-safety.

TheE committed:
Removed unnecessary db-call.

TheE committed:
Fixed wrong page number in error-message,

TheE committed:
Fixed two syntax errors in the localization files.

TheE committed:
Pushed bukkit to 1.7.2-R0.1 in pom.xml

TheE committed:
Merge branch 'master' into per-world-limits

TheE committed:
Made /warp assets display the warps per limit

TheE committed:
Removed debug output

TheE committed:
Pushed bukkit to 1.7.2-R0.2 in pom.xml

TheE committed:
Remove all cached resource bundles when unloading.

TheE committed:
Added missing java docs

TheE committed:
Refactored warp-limits to be more understandable.

TheE committed:
Fixed /warp assets for overlapping limits
While I do not see why anybody would want to have two limits overlapping
each other it is possible (which means also accidentally possible) and
therefore /warp assets should show the situation as it is configured.

TheE committed:
Fixed syntax error within warp-creation messages

TheE committed:
Removed some lines that were already commented out

TheE committed:
Merge branch 'per-world-limits'

TheE committed:
Rewrote /warp welcome to allow complete removal of a welcome message

TheE committed:
Also check for localization-files that use '.yaml' as ending.

TheE committed:
Show users the given answer when setting welcome messages.

TheE committed:
Fixed some welcome related messages.

TheE committed:
Renamed localization files to be unique and removed an unneeded method

TheE committed:
[maven-release-plugin] prepare release 2.6.4

TheE committed:
[maven-release-plugin] prepare for next development iteration

TheE committed:
Check if a MySQL connection is valid before using it.
Sadly, the SQLite driver used by bukkit does not implement the isValid()
method, so adding a connection check for SQlite is not as easy.

TheE committed:
Made sure config-files are saved with file encoding if forced.

TheE committed:
Fixed per-world limit-disobey permissions
If a user can disobey limits on one world out of a multi-worlds limit,
this world should be completely ignored when determining if the player
can build warps on an other world out of this limit.

TheE committed:
Fixed all missing or wrong parameter informations in java-docs.

TheE committed:
Fixed give description key and a typo in the German localization.

TheE committed:
Catch MissingResourceExceptions, return the key and log a warning.

TheE committed:
Fixed NPE when using '/warp assets' with zero warps

TheE committed:
Fixed value order in German invitation message.

TheE committed:
Pushed maven compiler plugin to the newest version.

TheE committed:
Pushed bukkit to 1.7.2-R0.3 in pom.xml

TheE committed:
Use a '&' as colour char and added a note about file encoding.
Localization files that use the '§' do still work without any
additionally changes on MyWarp's side as Bukkit just changes the '&' to
a '§' internally. Using the '&' here has the benefit that it can be
typed in-game.

TheE committed:
Exclude all non-usable commands from the help

TheE committed:
Call Java Plugin's constructor before doing anything.

TheE committed:
Rewrote '/warp assets', overwritten limits are now ignored

TheE committed:
Using Guava's Table to store running Timers avoids the former mess

TheE committed:
Pushed bukkit to 1.7.5-R0.1-SNAPSHOT in pom.xml

TheE committed:
Made Warmup and Cooldown store the player via UUID

TheE committed:
Rewrote MyWarp's core from sketch to support UUIDs
The whole inner logic has been completely rewritten to use Guava's
collection utils to simplify access and get rid of all ugly access
methods.
This compiles, but the data connections are basically just a hack to
connect MyWarp's new logic with the old database scheme and should never
be used in a productive environment.
Java docs will be added later on.

TheE committed:
Use the current Vault repository instead of the old one

TheE committed:
Moved all permission dependencies into their own repository

TheE committed:
[maven-release-plugin] prepare release 2.6.5

TheE committed:
[maven-release-plugin] prepare for next development iteration

TheE committed:
Pushed bukkit to 1.7.8-R0.1-SNAPSHOT in pom.xml

TheE committed:
Updated Bukkit to 1.7.9-R0.1 in pom.xml

TheE committed:
Updated database layer for UUID support using JOOQ for database access
This commit adds the missing support for UUIDs on the database side by
introducing a completely new database layout. Old databases are not
converted automatically, but need to be imported via the '/warp import'
command using the 'legacy' options.
Additionally, both Migrators and DataConnections are now handled
asynchronous if the implementation requires it - similar to the design
on the 'fully-async-dataconnections' branch.
The implementation works but should not yet considered stable! Javadocs
will be added later on.

TheE committed:
Fixed possible exceptions when using '/warp import'
Ugly, but it works.

TheE committed:
Update Vault dependency to use VaultAPI 1.4, addresses #7.

TheE committed:
Removed warp type symbol.

TheE committed:
Fixed second placeholder in commands.private.privatized message

TheE committed:
Cleaned up library code, Guava is preferred over Apache Commons.
'MinecraftFontWidthCalculator' has been renamed to 'FormattingUtils' and
the documentation has been updated. Methods have been cleaned-up to fix
#9.

TheE committed:
Fixed major bugs in the JOOQConnection

TheE committed:
BRAKING CHANGE: Rewrote permission value bundles completely.

This commit introduces a completely redesigned handling of
permission-values that is a lot more flexible and much more
object-oriented than previous attempts: 

- Limit Bundles: The complete evaluation of limits is now stored within
the limit-bundle. The multiworld-approach has been moved into abstract
classes to be reusable.

- Fee Bundles: The fee bundle includes a connection with the current
economy link to simplify the whole process. The economy-link has been
split into a ManagedLink that includes player-messages and
bypass-permissions, and the actual link to the (external) economy
provider. The permission to disobey all fees has changed to
'mywarp.economy.disobey'.

- Time Bundles: Warp warmups and cooldowns have been moved into one
object, the time bundle. This also changes the configuration and
permissions for timers: player need to hold the
'mywarp.timer.[IDENTIFIER]' permission, the bypass permission has been
changed to 'mywarp.timer.disobey', 'mywarp.timer.disboey.dmgabort' and
'my warp.disobey.moveabort'. The management of TimerActions has been
split into several classes and should be pretty strait-forward.

The configuration itself will be updated later-one to properly adapt
this new style.

TheE committed:
Removed 'dependency-reduced-pom.xml'

It should not have been there in the first place.

TheE committed:
Delete dependency-reduced-pom.xml

TheE committed:
Fixed dynmap markers for now

TheE committed:
Informs users about too long warp names and failed player teleports,
renamed generated package

TheE committed:
Fixed economy disobey permission check

TheE committed:
Implemented warp creation dates
dates are implemented via Java's rather poor API. If further functions
are needed, Joda Time (or at some point Java 8) could be used without
changing the whole logic. The altered popularity algorithm is more or
less a shot in the dark and might need further tuning, especially via
the gravity constant.

TheE committed:
Minor optimization of the database migration

TheE committed:
BRAKING: Rewrote configuration handling
MyWarp now defines an interface that provides an abstraction-layer for
all needed values. The actual configuration handling is done by
implementing this interface.
The conversion of the old configuration file (properties) has been
removed for now.
Constructors taking ConfigurationSections in ValueBundles have been
removed, since they enforce a strict dependency on the configuration
handling mechanism that is no longer exeptable.
The configuration-file has been reformatted and users need to
reconfigure it.

TheE committed:
Pushed Bukkit to 1.7.9-R0.2 in pom.xml

TheE committed:
Removed '/warp accept' in favor of conversion API usage.

TheE committed:
Fixed '/warp assets' when limits are disabled, removed name null checks
The removed null-checks when converting the invited-player-ids to names
does not mean that this problem has been solved. Rather the problem
affects the whole plugin and needs to be solved globally. It has been
registered as #15.

TheE committed:
Fixed SQLite Migrator using MySQL dialect.

TheE committed:
Rewrote Markers to use one method for all updates, fixed dynmap for now
The rewritten interface puts control over the conditions to display a
markers for a warp completely in the implementation.
Remaining bugs in the dynmap implementation have been fixed, as far as
they are caused by MyWarp. There seems to be an issue with moving a
marker to a different world but that should not be caused by MyWarp.

TheE committed:
Moved standard predicates for warps into an utility class.

TheE committed:
First attempt to clean-up the constructors of the warp class

TheE committed:
Added checkstyle to pom.xml
Lots of minor things have been changed all over the plugin in order to
fix violations of the defined checkstyle rules. The UUIDFetcher is
currently not covered since it will be removed soon due to licensing
problems.
Additionally, JOOQ is now shaded into an internal package and the pom
has been reformatted.

TheE committed:
Removed PaginatedResult in favor of a new, modular StringPaginator

TheE committed:
Integrate flyway for database migrations
At the moment the integration covers future migrations only, as
migration of legacy layouts is still handled by the custom migrators.
Using the automatic flyway migration could enforce uneeded dependencies
on old propperties (e.g. table-name) that are needed for the conversion,
but not for any new behaviour.
Additionally handling the database could be problematic as the new
layout should have its own database in most configurations - the old,
single table layout however could have easily been placed in a database
common for minecraft plugins (as was indicated in the default
configuration).

Fixes #16.

TheE committed:
Remove renaming of legacy 'homes-warps.db'
The database format is no longer supported, so there is no point in
renaming it. If users still have this database around, user could enter
its name in the import command.

TheE committed:
Create data folder if it does not exist

Fixes #19

TheE committed:
BREAKING: Split core and plattform implementation
With this commit MyWarp has a platform independent core that does not
use any platform specific code, and a platform specific implementation
that adapts this core to the logic used by the platform running it. As
of this commit, only Bukkit is supported and the implementation is not
yet splitted into another Maven module.

There are also some functional differences that should be noted:
- The names of the SQL tables, columns and constraints have changed.
- The world is now identified and stored by UUID.
- Additional logic to lookup player-names have been added on a per-plattform-level. Fixes #15.
- The creation-date of a Warp is now stored in the database and used by
the popularity algorithm, this fixes #10.
- The internationalization has been rewritten and uses standard
Properties resource bundles. For now they are not accessible to the
end-user.
- Formatting has been completely removed from the internationalized
resources and is hard-coded to make resources platform independent.
- The custom command management has been replaced by Intake, this fixes #11.
- The ‚-p‘ flag for list and search is gone.
- The list command has been altered, its ‚-c‘ and ‚-w‘ flags check if
the given query is contained in the name of the creator or world.
- A simple internal event system for warps has been added that uses
Guava’s EventBus to notify subscribers of changes. It is currently used
to notify that data layer and Dynmap, if enabled.
- Each class that needs to log something now uses its own logger. These
loggers currently do not append the plugin’s name as they probably
should.

There also several aspects that still need work. Most of them are
documented right in the source. It is fully expected that the overall
design will change again as there are still some problems left to solve.

This commit has only been briefly tested. Standard functions work, but
it is expected that there are bugs or unintended behavior. By no means
this is a stable product!

TheE committed:
Remove the search commend, add '-n' to the list command.
Fixes #21.

TheE committed:
Switch to gradle
Additionally the checkstyle rule 'RedundantThrows' has been removed as it produces false positives and is deprecated. readme and gitignore has been updated to reflect the changes.

TheE committed:
Fix gradle rules in gitignore, convert to  Google Code Style
Some Checkstyle rules are still not meat, but this will be addressed soon.

TheE committed:
Add missing Gradle wrapper

TheE committed:
Fix CONTRIBUTING.MD formatting

TheE committed:
Tame Checkstyle, fix formatting

TheE committed:
Remove static MyWarp singleton instance

TheE committed:
Fix issues with the Bukkit version

TheE committed:
Add .gradle to gitignore

TheE committed:
Remove hardcoded 'mywarp' database name in Flyway

TheE committed:
Fix wrong translation message in ImportCommands

TheE committed:
Fix NPE in LegacyMigrator and readded asyncronus migrations
The legacy migration code is still terrible slow, expecially the UUID lookup which request one single UUID after another is ugly and might fail for databases with higher player numbers. This should be completly rewritten.

TheE committed:
Fix page number logic and missing translation in StringPaginator

TheE committed:
Fix UnsupportedOperationException when checking multiple limit types

TheE committed:
Add a token to display warp information to WarpAcceptanceConversation

TheE committed:
Fix listing of warps without creator name or world when using '-c' or '-w'

TheE committed:
Add bulk name to unique ID conversion and implement SQLite profile cache for Bukkit

TheE committed:
Merge branch 'uuid-changes'

Conflicts:
	src/main/java/me/taylorkelly/mywarp/MyWarp.java

TheE committed:
Remove leftover debug stacktrace

TheE committed:
Add Travis.

TheE committed:
Add 'r' to the list command to only display warps within a given radius
Fixes #13.

TheE committed:
Remove leftover search value in the Bukkit configuration file

TheE committed:
Fix NPE when using the assets commit

TheE committed:
Fix wrong permission creation for ValueBundles

TheE committed:
Fix formatting of lists
Fixes #25.

TheE committed:
Use new localization support in Intake for dynamic localizations

TheE committed:
Fix remaining javadoc and checkstyle issues

TheE committed:
Properly implement InvalidUsageException handling

TheE committed:
Clean up Bindings, use classifiers where necessary

TheE committed:
Add dummy help implementation

TheE committed:
Fix confusing parameter order in LegacyMySqlMigrator

TheE committed:
Use a custom binding to resolve SQLite db files

TheE committed:
Change file-not-found message key accordingly

TheE committed:
Remove all 'NON-NLS' comments
There are better ways to ignore non-localized strings, like IntellJ Idea's external 'Non-Nls' annotation.

TheE committed:
Add missing translation for invalid syntax in WarpDispatcher

TheE committed:
Create base localization files automatically upon build
Fixes #8.

TheE committed:
Update shadow plugin to fix dependency resource relocation

TheE committed:
Update for latest Intake version, move Intake resources into Bukkit module

TheE committed:
Add German localization

TheE committed:
Unify DynamicMessages javadocs

TheE committed:
Implement help

TheE committed:
Fix ResourceBundle encoding issues
All ResourceBundles are encoded in UTF-8 for simplicity. This is actually implemented in the FolderSourcedControl in the code module, but this remains unused for know. The new EncodedControl acts as a bridge until the FolderSourcedControl is implemented (if it ever is).

TheE committed:
POTENTIALLY BREAKING: Fix massive errors with UUID storage
This commit changes the way UUIDs are stored in both MySQL and SQLite from BINARY(16) to VARCHAR(36). This corresponds to the way JOOQ emulates UUID support for both databases. While the nature of this migration will normally result in data lose, this should not be an issue here:
- For MySQL, since the first 'official' available devBuild (commit 60c5d26ad724858e8a1407ca10a919af3477d1e8), 'worlds.uuid' had the BINARY(16) type but was actually treated as if it were a VARCHAR(36) column in the implementation. Therefore no worlds could be stored. `players.uuid' was treated as BINARY(16) until ad0ee68447a92c54a963ef9eb2604b006ad37430 that changed the internal treatment to expect VARCHAR(32). However, since the Bukkit version did not actually start until it was fixed two commits after that one, there should not be anyone with a MySQL database that actually contains players.
- For SQLite the same internal problems apply. But because of SQLite's unstrict type handling, values where actually written and read as VARCHAR(32) instead of failing. Since data could only be there since ad0ee68447a92c54a963ef9eb2604b006ad37430 (which started the VARCHAR(32) problem for players), every UUID within SQLite databases is stored as VARCHAR(32). Therefore there is no need to convert the data, it just needs to be preserved.

TheE committed:
Fix missing invitation storage when adding or importing a warp

TheE committed:
Push JOOQ to 3.5.3.

TheE committed:
Fix '/warp' permission and usage with multiple word warp names
Requires the latest Intake commit.

TheE committed:
Fix broken build caused by imports (?)

TheE committed:
Made the 'to' command usable under Bukkit.

TheE committed:
Fix warp signs.

TheE committed:
Move Dynmap configuration entirely to BukkitSettings and remove unused nodes

TheE committed:
Fix inverted logic for yaw and pitch

TheE committed:
POTENTIALLY BRAKING: Change permission group prefix to 'group.[GROUPNAME]'
Also the PermissionsEx handler is removed. In fact, PEX automatically adds such a group permission for each group, so the SuperPermsHandler can be used. Other handlers (GroupManager and bPermissions2) internally support UUIDs but do not provide an API that accepts UUIDs. For now these handlers are left in place, but if there are any problems they will be removed.
Fix #7.

TheE committed:
Check warp names against 'warp' command aliases
Fix #32.

TheE committed:
Add a custom logger that delegates calls to slf4j and adds a prefix
Usage of slf4j allows to use adapt the actual logging to the logging framework used by the platform. On Bukkit the logging hooks directly into Minecraft's log4j.

Fixes #31.

TheE committed:
Remove local maven repo.

TheE committed:
Show the SQL exception when migration fails

TheE committed:
Fix missing value in invitation message

TheE committed:
Reference commit, ci and build number in human readable version and show them in the Manifest
The information are given to the build process via environment variables, typically via Gradle's '-P' option. If the variable is missing, a default is used instead. Actually used are the following:
- 'buildNumber'
- 'ciSystem'
- 'commit'

 For Travis, the information are not (yet) added.

TheE committed:
Use Grgit to resolve the commit hash
Apparently Jenkins Gradle plugin cannot shorten the git commit hash via Token Macro Plugin like it should.

TheE committed:
Fix invitation message once again.

TheE committed:
Move Locale resolving to its own class

TheE committed:
Fix reflection logic for getting a player's Locale

TheE committed:
Add 'f' flag for public and private command to ignore limits of the warp owner
This change also allows to give warps to users who are offline.

TheE committed:
Rename 'Condition' to 'Name' and 'Condition.Type' to 'Name.Condition'

TheE committed:
Distinguish between limits exceeded by somebody else and limits exceeded by the initiator

TheE committed:
BREAKING: Overhaul permissions
With this commit nearly all permissions have been changed. The configuration is also updated.

This step was necessary since the old 'bag' system (e.g. 'mywarp.warp.basic.*') included permissions, that unlocked behavior because flag permissions or permission for usage with other players where included. Besides, the new permission structure is ready for adaptin of systems with implicit permission inheritance (Sponge) and feels a lot more intuitive.

TheE committed:
Push JOOQ to 3.5.4

TheE committed:
Support building with Java8, update Gradle wrapper to 2.3

TheE committed:
Refactor timers and CommandUtils into the core, parametric stuff into its own package

TheE committed:
Add change log

TheE committed:
Release 3.0-beta-1.

TheE committed:
Prepare for next development iteration

TheE committed:
Fix '/warp player' being unusable by non-players.

TheE committed:
Deploy javadoc to GitHub pages via travis on each successful build
Fix #12.

TheE committed:
Fix missing multi module configuration for javadoc deployment

TheE committed:
Attempt to fix javadoc deployment

TheE committed:
Attempt to fix javadoc deployment script permissions

TheE committed:
Attempt to fix javadoc deployment script again

TheE committed:
Attempt to fix javadoc deployment script again

TheE committed:
Fix Travis environment variable

TheE committed:
Fix javadoc deployment path

TheE committed:
Fixed NPE thrown when formatting special characters

TheE committed:
Remove unneeded Locale parameter in PaginatedResults

TheE committed:
Update javadoc publishing for recent site changes

TheE committed:
Silence the cloning command.

TheE commented:
Err, I assume you wanted to create an issue for my repo and not a pull request for mung3r's (which my is based on). Still, let me answer your questions.

You mention three separated problems:
- Tags instead of `{0}`: Localization worked like this when I first implemented them - and it was a mess. There where lots of bugs where I simply had either the wrong tag or a simple typo in the tag right in the code, that messed up the localization. With the current system, there are no such duplicates and such erros can no longer happen. Besides I ma using Java's `MessageFormat` witch handles thinks like locale specific formatting of numbers and dates automatically - once again, less stuff that could potentially go wrong. So no, I am keeping the current syntax.
- Requirement of `'`: The large part is caused by the YAML syntax and the stupid implementation of Bukkit's configuration parser. This is gone in MyWarp3 with uses Java Properties files (see [here](https://github.com/TheE/MyWarp/blob/master/mywarp-core/src/main/resources/me/taylorkelly/mywarp/lang/Commands_en.properties) for an example).
- Hard coded formatting codes such as `&c`: You are correct. This is why I completely removed them in MyWarp3 and hard coded all formatting.

Remski01 commented:
Oke then not :p 
But yea thanks for answer. I understund it :)

Have an great day/night

TheE committed:
Fix '/warp assets' being empty for players without warps and invalid variables in German localization

TheE committed:
Overhaul WarpManager implementation, block server shutdown until SQL executor is shutdown
The WarpManager implementation uses the Decorator pattern to add new functions to the HashMap backed in memory implementation. Most of this implementation logic is hidden and - for know - only internal behavior. Additionally the WarpBuilder has been rewritten and cleaned up.

The Executor used for SQL execution now blocks the main thread when shutting down - either for 30 seconds, or until all submitted tasks have been executed. This addresses #43.

TheE committed:
Fix missing wrapping populated of persistent and eventful warps

TheE committed:
Cascade foreign keys and use SQLite's foreign key support.
Addresses #43.

TheE committed:
Fix typo in javadoc deployment script.

TheE committed:
Let platforms specific a DataSource to be used by the core
This commit also changes larger parts in DataConnection logic and interfaces:
- The DataConnection interface no longer makes any enforcements about multi-threading logic. Likewise, multi-threading code has been removed from the JOOQConnection. Async writing is handled by the AsyncWritingDataConnection class that can be implemented where needed.
- DataMigrator has been removed in favor of WarpSource. This interface no longer makes any enforcements about multi-threading behavior and is implemented natively by DataConnection. LegacyWarpSource converts legacy databases from a given DataSource.
- DataConnections are acquired by the corresponding factory that also handles FlyWay initialization where needed.
- All database settings have been removed from the Settings interface and are exclusively defined on platform level.
- On Bukkit, in absence of a native SQL pool, MySQL and SQLite use the SingleConnectionDataSource that shares a single Connection. DataSources are created via the corresponding factory. This emulates behavior as it was har-coded at core level until this commit.
- Rewritten multi-threading code is compatible with Guava10 and 18 (no more usage of Futures.chain(ListenableFuture, Function)).

Closes #29, #46.

TheE committed:
Overhaul JooqConnection implementation: Remove Record CRUD, use transactions
While the new implementation avoids a lot of unnecessary queries and works fine with current MySQL and SQLite versions, it is not usable on a current unmodified CraftBukkit/Spigot server. Both bundle sqlite-jdbc 3.7.2 that does not yet implement transactions. Future commits will need to solve this problem.

TheE committed:
Rename 'DataConnection' to 'WarpStorage' and modify corresponding classes and packages accordingly

TheE committed:
Fix getEconomyService() always throwing UnsupportedOpperationExceptions on Bukkit
Fixes #47.

TheE committed:
Fix economy failure for positive amounts, overhaul EconomyManager definition and JavaDocs
Core interfaces now use BigDecimal internally to eliminate possible rounding issues.

TheE committed:
Fix irregular handling of AuthorizationExceptions in FallbackDispatcher
Fixes #49.

TheE committed:
BREAKING: Add support for H2 and use it as default. Require at least SQLite 3.8.7.2
This commit once again introduces several structural changes:
- RelationalDataService implementations are required to bundle a ConnectionConfiguration that wraps information about features supported by the underling database management system. These information are used when initializing table structure and using RelationalWarpStorage.
- FlyWay creates the configured database schema if it does not yet exist. All database creation scripts have been altered to add a schema placeholder where necessary and where reformatted. Checksums are updated automatically in existing setups.
- JOOQ renders queries with the full schema and only omits it if the underling database does not support schemas.
- WarpSource and implementations have been moved into the 'storage' package to profit from package private utilities. LegacyWarpSource also uses the new ConnectionConfiguration.
- On Bukkit the MySQL configuration has been replaced by a more flexible storage configuration that requires a JDBC URL, a schema (if supported) and user + password (if supported) to connect to the database.
- Existing import commands to import from up-to-date databases have been replaced by a single command that requires a JDBC URL, a schema (if supported) and user + password (if supported).
- Internally ConnectionConfiguration is used to create dataConnections instead of raw strings.

SQLite is still supported, but at least version 3.8.7.2 of the JDBC driver is required. Bukkit bundles an older version; it is therefore impossible to use an SQLite database on Bukkit! Instead H2 is used by default, existing SQLite data needs to be imported.

Fixes #48.
Addresses #30.

TheE committed:
Merge branch 'per-platform-datasource'

# Conflicts:
#	mywarp-core/src/main/java/me/taylorkelly/mywarp/MyWarp.java

TheE committed:
Add 'getParentLogger()' method to SingleConnectionDataSource to fix Java7+ builds

TheE committed:
Allow addition of external new resource bundles as well as replacement of existing ones
- Users may override bundled resource bundles or add entirely new ones by putting them in the 'lang' folder inside of MyWarp's plugin folder. Bundles must be placed in sub-folders that mirror MyWarp's package structure and named accordingly (e.g. 'me/taylorkelly/mywarp/lang/Commands_de.properties' overwrites the bundled german bundle).
- Custom files will not be updated in any way, missing keys are resolved via the standard bundle lookup order.
- The root resource bundle cannot be overwritten.
- Folder and files will not be created automatically. By default only bundled files will be used.

Adds #33.

TheE committed:
Replace 'TeleportService' with 'TeleportManager', clean up implementation requirements
Implementation details of the TeleportManager like the safety algorithm are hidden at a package level and can be modified without risking API inconsistencies.

TheE committed:
Move tokens into WarpUtils.

TheE committed:
Update readme with localization information.

TheE committed:
Sort and unify new lines in localization files

TheE committed:
Add support for MariaDB.
Since MariaDB ensures a "drop-in" replacement capability with MySQL, the migration scripts of MySQL are reused.
Fixes #50.

TheE committed:
Update changelog with latest changes.

TheE committed:
Release 3.0-beta-2.

TheE committed:
Prepare for next development iteration

TheE committed:
Validate connections before using them
Fixes #58.

TheE committed:
Fix ownership changes of a single warp affecting all warps of a player
Fixes #53.

TheE committed:
Update changelog with latest changes.

TheE committed:
Release 3.0-beta-3.

TheE committed:
Prepare for next development iteration

TheE committed:
Fix broken diff links in changelog

TheE committed:
Move authentication from Warp into its own service
With a dedicated service, authentication should be more flexible than before. At some point, the API might allow clients to switch the used strategy and therefore completely customize the plugin's behavior. As of this commit, everything should behave as it did before, though.
Additionally some minor changes have been made within the class structure: AbstractWarp now contains several interchangeable methods from Warp and AbstractPlayer is gone as it is no longer needed.
Addresses #64.

TheE committed:
Push JOOQ dependency to 3.6.4
This is the latest version that still supports Java6,  3.7 (and onwards) only support Java8.

TheE committed:
Rename several unclear classes, make TeleportService implement the strategy pattern
Also several javadocs where updated and reformatted.
Closes #66.

TheE committed:
Remove unneeded methods from Warp

TheE committed:
Fix damage and move abort for warp warmups
Fixes #81, #78 (?)

TheE committed:
Make DynamicMessages use bundled resources by default, decouple FolderSourcedControl

TheE committed:
Update copyright for 2016

TheE committed:
Add support for gold and iron pressure plates
Adds #77

TheE committed:
Move commands into core, add message system for platform independent message formatting

TheE committed:
Rename WarpRemoval Event to WarpDeleteEvent to match internal naming

TheE committed:
Move teleport from Warp into TeleportServices
A basic TeleportService implementation can be acquired from MyWarp. Further checks (like Timers or Economy) can be implemented on top by the caller.

TheE committed:
Update to Intake 4

TheE committed:
Require JavaDocs on all protected and public methods and classes only
From now on javadocs on all private or package private methods and classes are optional. Most of the existing ones are meaningless and take to much time to update - they might be removed in future versions.

TheE committed:
Massive refactoring, add platform capabilities, overhaul Bukkit implementation

This commit moves almost every existing class into a new package structure that more closely represents the internal design of the project. It also introduces a clear split between service and capability:
- A capability is provided by the platform and requested by the platform when needed. Capabilities are generally optional, the core will simply ignore features the platform cannot provide capabilities for.
- A service is an abstraction over a capability. Services are created locally by interaction models that require them. Every interaction model must therefore adapt itself to missing capabilities.

As a result of this split, the Bukkit implementation has been completely overhauled. Notability the dynmap support has also been rewritten and no longer enforces any implications on the warp's displayed (see #63).

Fixes #65.

TheE committed:
Remove 'getMatchingWarps(String, Predicate)' from WarpManager

TheE committed:
Replace MyWarp constructor with factory, replace core events with platform methods

The events might be reintroduced if there is reason to believe clients or API user might need to be notified about reloads. For now platform methods are simply easier to use.

TheE committed:
Move handling of tamed horses into core

TheE committed:
Fix resource bundle location on Bukkit, create english defaults for all sub-projects

TheE committed:
Update Flyway to 4.0

TheE committed:
Add previously missing conversation resource bundles

TheE committed:
Remove unnecessary copyright statements form resource bundles

TheE committed:
Improve core API for warp signs

TheE committed:
Remove now unnecessary line overwrite

TheE committed:
Fail quietly if the client returns an invalid locale

TheE committed:
Update profile cache asynchronous on login
Fixes #98.

TheE committed:
Fix non-working checks for warp-creation
Fixes #99

TheE committed:
Fix error message

TheE committed:
Fix inverted checks for already existing warps

TheE committed:
Make teleport safety a platform capability

Theoretically, different MaterialInfo instances could be used for different versions of Minecraft or Bukkit. There probably will be some more changes  regarding the way individual blocks are handled by the core, but this remains to be seen once support for physical warp signs is moved to the core.

Closes #85, addresses #95.

TheE committed:
Fix NPE when converting ArgumentExceptions without a cause

Fixes #106.

TheE committed:
Add Tab complete for Warps, LocalPlayers and Profiles

In terms of functionality, this commit is pretty much experimental: Warps, LocalPlayers and Profiles all use the matching logic that was previously used to parse warp names (see documentation of Matcher for details). Suggestions on Tab are ordered alphabetically (LocalPlayers, Profiles) or by popularity (Warps). Due to the behavior of Intake, Tab complete currently does not work for warps when the '/mywarp <warp>' command is used as it parses the sub-commands instead.

This commit requires that latest version of my Intake fork. Use '--refresh-dependencies' if the build fails.

Closes #107, #83.

TheE committed:
Fix creation of the plugin folder on Bukkit

Closes #104.

TheE committed:
Fix IndexOutOfBoundsException when matching warp names

TheE committed:
Replace EulerDirection with vectors

TheE committed:
Update FlyWay to 4.0.2 and fix copyright formatting
Closes #108.

TheE committed:
Clean up resource bundles

Resources for timers are finally stored in their own bundle; bundles for warp signs and paginated content have been renamed. Several localization keys have als been renamed in an attempt to unify them.

These changes are already present on crowdin.

Addresses #109.

TheE committed:
Clean up build files, require at least Java 8 to run the build

TheE committed:
Add automatic download of translations from crowdin upon build

To download the translations, crowdin's API key is required. It can be set when launching the build by using the 'crowdinKey' property like so: 'gradlew clean build -PcrowdinKey=PRIVATE-API-KEY'. If this property is not set, the build continues with the bundled translations only.

The German translation is no longer bundled directly by MyWarp as it is hosted on crowdin and bundled upon build.

Fixes #109.

TheE committed:
Fix a possible crash for malformatted value bundles, use correct database path when falling back to defaults

If a value bundle configuration is not a valid ConfigurationSection, the values are ignored and a warning is logged. If a value bundle's values are missing, defaults are used.

If MyWarp fails to create the configuration file and falls back to build-in defaults, the default path to the H2 database is now used correctly.

Fixes #102.

TheE committed:
Move support for physical warp signs into the core

As before, implementations are responsible for listening to the relevant events and filtering these events for relevant interactions. These interactions are than given to the core to handle them, that is to search for warp signs associated with them.
While this implementations requires API users to manage relevant interactions - that is interactions with relevant blocks - by themselves, this should be faster, especially on platforms like Sponge that allow some native event filtering based on involved blocks.

Fixes #84.

TheE committed:
Remove direct of Limit.Type form Warp.Type

This method is still pretty much magic and brakes the predicates that are used elsewhere, but for now it's better than before.

TheE committed:
Remove Profile, directly convert UUIDs to Strings when needed

 Since the name of of a certain unique identifier can change while MyWarp is running, there is no other way to implement Profiles than to always query the name when it is needed. This however eliminates the need for Profiles altogether since this conversion can be made explicitly.

 There are still some areas like exceptions where currently UUIDs are displayed.

TheE committed:
Fix non working teleport effect setting, remove chuck loading setting

The configuration to disable chuck loading was never implemented in version 3 and on Bukkit it is better to enforce it. On other platforms however it should never be needed so it can be implemented for Bukkit only.

TheE committed:
Use events to inform players about invitations

TheE committed:
Fix 'warp assets' for limits of zero

TheE committed:
Update with new official repo for VaultAPI

See MilkBowl/VaultAPI#24

TheE committed:
Update changelog with latest changes

TheE committed:
Move (de)population from WarpManager into a separated (hidden) interface

TheE committed:
Handle warps with non-existing worlds more gracefully

Before this commit, trying to get a no longer existing LocalWorld from a Warp resulted in a RuntimeException that was or was not handled, depending on the interaction model. As of this commit, the callers need to convert the stored world identifier to a LocalWorld by themselves, giving them control over the response if the world does not exist:
- Teleports to warps with non-exting worlds are apported and the user is informed accordingly.
- The 'info' and 'list' commands both display the stored UUID if the world does not exist.
- The 'give', 'public' and 'private' commands fail if the warp's world is unavailble and limits must checked (limits can only be checked for existing worlds).
- If world access control is enabled, Warps without an existing world cannot be viewed or used (if the world does not exist, there is no way to check if the user would be able to use the warp).
- If Dynmap is used, warps with non-exting worlds are ignored.

Additionally the handling of world access control and permission based authentication has been changed. Before this commit, world access control could overrule permission overrides; as of this commit, permission overrides overrule world access control.

Fixes #93.

mung3r commented:
Merge request on wrong fork.
